Test/VimWpfTest/CommandMarginUtilTest.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Linq;
+using Microsoft.FSharp.Collections;
+using Moq;
+using Vim.EditorHost;
+using Vim.UI.Wpf.Implementation.CommandMargin;
+using Vim.UnitTest;
+using Vim.UnitTest.Mock;
+using Xunit;
+
+namespace Vim.UI.Wpf.UnitTest
+{
+    public class CommandMarginUtilTest : VimTestBase
+    {
+        private readonly MockRepository _factory;
+        private readonly Mock<IIncrementalSearch> _search;
+        private readonly MockVimBuffer _vimBuffer;
+
+        public CommandMarginUtilTest()

```

Comment:
Wow, you really got the full-blown experience by cooking up one of these test harnesses from scratch!